### PR TITLE
DCD-380: Add Aurora DB support

### PR DESCRIPTION
**Pull request by bpar476**
See #12 for full discussion, the git history on the branch somehow got corrupted 😕

### CODE REVIEW

File: templates/quickstart-jira-dc.template.yaml [hunk 1]
```diff
@@ -296,39 +299,64 @@ Parameters:
...
+      - db.t3.large
+      - db.t3.xlarge
+      - db.t3.2xlarge
       - db.t2.medium
       - db.t2.large
       - db.t2.xlarge
       - db.t2.2xlarge
     ConstraintDescription: Must be a valid RDS instance class, from the selection list
-    Description: RDS instance type
+    Description: RDS instance type (must be r4 family if using Aurora)
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc.template.yaml]: Aurora reference must be commented

File: templates/quickstart-jira-dc-with-vpc.template.yaml [hunk 2]
```diff
@@ -333,39 +326,64 @@ Parameters:
...
+      - db.t3.large
+      - db.t3.xlarge
+      - db.t3.2xlarge
       - db.t2.medium
       - db.t2.large
       - db.t2.xlarge
       - db.t2.2xlarge
     ConstraintDescription: Must be a valid RDS instance class, from the selection list
-    Description: RDS instance type
+    Description: RDS instance type (must be r4 family if using Aurora)
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc-with-vpc.template.yaml]: Aurora reference must be commented

**Review by theghostwhoforks — CHANGES_REQUESTED** [review]

**Comment by bpar476** [comment]
I've addressed the changes you've requested and also bumped the Atlassian services version :D

File: ci/params/aurora/quickstart-jira-dc-aurora-params.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  {
+    "ParameterKey": "QSS3KeyPrefix",
+    "ParameterValue": "quickstart-atlassian-jira/"
+  },
+  {
+    "ParameterKey": "CidrBlock",
+    "ParameterValue": "10.0.0.0/16"
+  },
+  {
```
Comment (nanux) [inline ci/params/aurora/quickstart-jira-dc-aurora-params.json:26]: This is duplicated

**Review by nanux — APPROVED** [review]
Were some of the non-Aurora related changes blocking some other task? If that would be the case, I would need to reconsider the way how I should structure PR in future.

**Comment by bpar476** [comment]
> Were some of the non-Aurora related changes blocking some other task? If that would be the case, I would need to reconsider the way how I should structure PR in future.

No, we just wanted to get these changes in before any other changes to the quickstart were made so we wouldn't need to do a three way merge again, and we can just enable Aurora when we are ready to go-live

File: templates/quickstart-jira-dc-with-vpc.template.yaml [hunk 4]
```diff
@@ -420,7 +438,7 @@ Parameters:
     Type: String
   DBStorage:
     Default: 200
-    Description: Database allocated storage size, in gigabytes (GB)
+    Description: Database allocated storage size, in gigabytes (GB). If you choose Provisioned IOPS, storage should be between 100 and 6144. # (not used for Aurora)
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc-with-vpc.template.yaml:441]: Do we have to change this description until we are ready to merge aurora?

File: templates/quickstart-jira-dc-with-vpc.template.yaml [hunk 5]
```diff
@@ -435,7 +453,7 @@ Parameters:
       - General Purpose (SSD)
       - Provisioned IOPS
     ConstraintDescription: Must be 'General Purpose (SSD)' or 'Provisioned IOPS'.
-    Description: Database storage type
+    Description: Database storage type #(not used for Aurora)
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc-with-vpc.template.yaml:456]: Do we need to comment out Aurora here until we are ready to support it?

File: templates/quickstart-jira-dc-with-vpc.template.yaml [hunk 6]
```diff
@@ -333,39 +326,64 @@ Parameters:
...
+      - db.t3.large
+      - db.t3.xlarge
+      - db.t3.2xlarge
       - db.t2.medium
       - db.t2.large
       - db.t2.xlarge
       - db.t2.2xlarge
     ConstraintDescription: Must be a valid RDS instance class, from the selection list
-    Description: RDS instance type
+    Description: RDS instance type # (must be r family if using Aurora)
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc-with-vpc.template.yaml:374]: Do we need to comment out Aurora here until we are ready to support it?

File: templates/quickstart-jira-dc-with-vpc.template.yaml [hunk 7]
```diff
@@ -333,39 +326,64 @@ Parameters:
     Default: ""
     Description: 'Use custom existing DNS name for your Data Center instance. This will take precedence over HostedZone. Please note: you must own the domain and configure it to point at the load balancer.'
     Type: String
+  # This parameter will be enabled when support for Aurora is added. Currently "dark" feature.
```
Comment (theghostwhoforks) [inline templates/quickstart-jira-dc-with-vpc.template.yaml:329]: Nit. We could make this a dropdown of 1 item, comment out the Aurora option and retain the dBEngine parameter, perhaps?

**Review by theghostwhoforks — CHANGES_REQUESTED** [review]

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 4 above)
Comment (bpar476) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:441]: Why would we change the description? The Aurora part is commented out

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 5 above)
Comment (bpar476) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:456]: It is  isn't it?

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 7 above)
Comment (bpar476) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:329]: I think that's probably confusing for end users. They might think why is there a drop down with only one option

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 6 above)
Comment (bpar476) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:374]: It is isn't it?

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 7 above)
Comment (nanux) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:329]: I agree, right now it is less confusing and adding/uncommenting Aurora related changes will be quick and we have a ticket to track it. I think we can merge this.

File: templates/quickstart-jira-dc-with-vpc.template.yaml (hunk 6 above)
Comment (theghostwhoforks) [reply templates/quickstart-jira-dc-with-vpc.template.yaml:374]: Misread the comment

**Review by theghostwhoforks — APPROVED** [review]

**Pull request merged**